presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDistributedQueriesWithOptimizedRepartitioning.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.presto.testing.MaterializedResult;
+import com.facebook.presto.tests.AbstractTestDistributedQueries;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static com.facebook.presto.sql.tree.ExplainType.Type.LOGICAL;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static io.airlift.tpch.TpchTable.getTables;
+import static org.testng.Assert.assertEquals;
+
+public class TestHiveDistributedQueriesWithOptimizedRepartitioning
+        extends AbstractTestDistributedQueries
+{
+    public TestHiveDistributedQueriesWithOptimizedRepartitioning()
+    {
+        super(() -> HiveQueryRunner.createQueryRunner(
+                getTables(),
+                ImmutableMap.of("experimental.optimized-repartitioning", "true",
```

Comment:
nit: 
```
ImmutableMap.of(
      "experimental.optimized-repartitioning", "true",
      "driver.max-page-partitioning-buffer-size", "200B"),
);
```

Also, 200B looks too small for partitioning buffer size?

Reply:
> Also, 200B looks too small for partitioning buffer size?

We deliberately chose a small page size because we wanted to force flush to happen. Since most of the queries in the tests are small we needed to choose a small number. We could of course have two tests, one with normal page size 1MB and another one with a smaller page size, but we didn't want to add too many tests. This test suite contains over 500 tests and take a long time to finish. Since the test with small pages covers the test with normal pages we just added the test with small pages. I added a comment in the code, how does it look?

```
          ImmutableMap.of(
                        "experimental.optimized-repartitioning", "true",
                        // Use small SerialiedPages to force flushing
                        "driver.max-page-partitioning-buffer-size", "200B"),
```

Reply:
@yingsu00 : This comment looks great! Thanks!